src/cozmo/robot.py

```diff
@@ -130,6 +130,67 @@ def _encode(self):
                                                 useManualSpeed=False,
                                                 usePreDockPose=False)
 
+class DockWithCube(action.Action):
+    '''Represents the dock with cube action in progress.
+
+    Returned by :meth:`~cozmo.robot.Robot.dock_with_cube`
+    '''
+    _action_type = _clad_to_engine_cozmo.RobotActionType.ALIGN_WITH_OBJECT
+
+    def __init__(self, obj, approach_angle, **kw):
+        super().__init__(**kw)
+        #: The object (e.g. an instance of :class:`cozmo.objects.LightCube`) that is being put down
+        self.obj = obj
+        if approach_angle is None:
+            self.useApproachAngle = False
+            self.approachAngle = util.degrees(0)
+        else:
+            self.useApproachAngle = True
+            self.approachAngle = approach_angle
+
+    def _repr_values(self):
+        return "object=%s" % (self.obj)
+
+    def _encode(self):
+        return _clad_to_engine_iface.AlignWithObject(objectID=self.obj.object_id,
+                                                     distanceFromMarker_mm=util.distance_mm(0).distance_mm,
+                                                     approachAngle_rad=self.approachAngle.radians,
+                                                     alignmentType=robotAlignment.RobotAlignmentTypes.Body.id,
+                                                     useApproachAngle=self.useApproachAngle,
+                                                     usePreDockPose=self.useApproachAngle,
+                                                     useManualSpeed=False)
+
+class RollCube(action.Action):
+    '''Represents the roll cube action in progress.
+
+    Returned by :meth:`~cozmo.robot.Robot.roll_cube`
+    '''
+
+    _action_type = _clad_to_engine_cozmo.RobotActionType.ROLL_OBJECT_LOW
+
+    def __init__(self, obj, approach_angle, check_for_object_on_top, **kw):
+        super().__init__(**kw)
+        #: The object (e.g. an instance of :class:`cozmo.objects.LightCube`) that is being put down
+        self.obj = obj
+        #: bool: whether to check if there is an object on top
+        self.check_for_object_on_top = check_for_object_on_top
+        if approach_angle is None:
+            self.useApproachAngle = False
```

Comment:
python variables should be snake-cased, so should be `self.use_approach_angle = False`